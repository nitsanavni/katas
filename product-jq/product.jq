reduce .[] as $a (null; $a[] as $e | (.//[]) + [$e]) // empty
