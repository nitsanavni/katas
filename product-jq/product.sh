jaq -n $1 | jaq -cf product.jq
