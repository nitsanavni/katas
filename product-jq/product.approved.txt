no input

single empty input array

single array with single item
[1]

single non-empty array
[1]
[2]

one array, more items
[1]
[2]
[3]

two arrays, one item
[1,2]

two arrays, two items
[1,3]
[1,4]
[2,3]
[2,4]

three arrays, one item
[1,2,3]

binary counter
[0,0,0,0]
[0,0,0,1]
[0,0,1,0]
[0,0,1,1]
[0,1,0,0]
[0,1,0,1]
[0,1,1,0]
[0,1,1,1]
[1,0,0,0]
[1,0,0,1]
[1,0,1,0]
[1,0,1,1]
[1,1,0,0]
[1,1,0,1]
[1,1,1,0]
[1,1,1,1]

